rotary/rules/rotary.smk

```diff
@@ -312,16 +369,222 @@ rule qc_long_length_stats:
+          minlength={params.min_read_length} stats={output.adapter_trim_stats} overwrite=t interleaved=f qin=33 \
+          pigz=t unpigz=t \
+          2> {log}
+        """
+
+
+rule short_read_quality_trimming:
+    """
+    Performs quality trimming after trimming adapters.
+    The conditional statements in the input field select whether to skip adapter trimming based on the config file.
+    Note: if quality_trim_direction is set as "f", the reads are just passed through this rule and not trimmed.
+    """
+    input:
+        short_r1 = "{sample}/qc/short/{sample}_adapter_trim_R1.fastq.gz" if str(config.get("perform_adapter_trimming")).lower() == 'true' else "{sample}/qc/short/{sample}_reformat_R1.fastq.gz",
+        short_r2 = "{sample}/qc/short/{sample}_adapter_trim_R2.fastq.gz" if str(config.get("perform_adapter_trimming")).lower() == 'true' else "{sample}/qc/short/{sample}_reformat_R2.fastq.gz"
+    output:
+        short_r1 = temp("{sample}/qc/short/{sample}_quality_trim_R1.fastq.gz"),
+        short_r2 = temp("{sample}/qc/short/{sample}_quality_trim_R2.fastq.gz"),
+        quality_histogram = "{sample}/qc/short/{sample}_quality_trim_qhist.tsv"
+    conda:
+        "../envs/qc.yaml"
+    log:
+        "{sample}/logs/qc/short/short_read_quality_trimming.log"
+    benchmark:
+        "{sample}/benchmarks/qc/short/short_read_quality_trimming.benchmark.txt"
+    params:
+        quality_trim_direction = config.get("quality_trim_direction"),
+        quality_trim_cutoff = config.get("quality_trim_score_cutoff"),
+        min_read_length = config.get("minimum_read_length_quality_trim"),
+        min_average_quality = config.get("minimum_average_quality_score_post_trim") if config.get("quality_trim_direction") != "f" else 0
+    threads:
+        config.get("threads", 1)
+    resources:
+        mem = config.get("memory")
+    shell:
+        """
+        bbduk.sh -Xmx{resources.mem}g threads={threads} in={input.short_r1} in2={input.short_r2} \
+          out={output.short_r1} out2={output.short_r2} qhist={output.quality_histogram} \
+          qtrim={params.quality_trim_direction} trimq={params.quality_trim_cutoff} minlength={params.min_read_length} \
+          minavgquality={params.min_average_quality} overwrite=t interleaved=f qin=33 pigz=t unpigz=t \
+          2> {log}
+        """
+
+
+rule short_read_contamination_filter:
+    """
+    Filters short reads based on match to a reference. 
+    Note that contamination_references in the input is only there to trigger download of the genome files; the actual 
+    text to be input into the shell (with proper comma separation) is prepared in all_contaminant_references in params.
+    """
+    input:
+        short_r1 = "{sample}/qc/short/{sample}_quality_trim_R1.fastq.gz",
+        short_r2 = "{sample}/qc/short/{sample}_quality_trim_R2.fastq.gz",
+        contamination_references = CONTAMINANT_REFERENCE_GENOMES_NCBI
```

Comment:
@jmtsuji Will this fail if CONTAMINANT_REFERENCE_GENOMES_NCBI = None? How will this work if there are no NCBI contamination references?

Reply:
Addressed in https://github.com/jmtsuji/rotary/pull/82